### PR TITLE
memoryPairGame_linkqwd

### PR DESCRIPTION
**Pull request by linkqwd**
### memoryPairGame

[Demo](https://linkqwd.github.io/piarGame/) |
[Code base](https://github.com/linkqwd/piarGame/blob/master/js/main.js)

### CODE REVIEW

File: submissions/linkqwd/task-10_memoryPairGame/main.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        }
+        cards.sort(() => 0.5 - Math.random());
+        document.querySelector(`.${DOM.cardsHolder}`).insertAdjacentHTML('beforeend', cards.join(''));
+    }
+
+    const makePreview = () => {
+        toggleClass(DOM.noClickCard);
+        setTimeout(() => toggleClass(DOM.openedCard), 500);
+        setTimeout(() => toggleClass(DOM.openedCard), settings.prewievDelay + 500);
+        setTimeout(() => toggleClass(DOM.noClickCard), settings.prewievDelay + 1000);
```
Comment (zonzujiro) [inline submissions/linkqwd/task-10_memoryPairGame/main.js]: Magic numbers are bad - don't use it.
Also, to chain these events, you can use promises.

File: submissions/linkqwd/task-10_memoryPairGame/main.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    }
+
+    const updateScores = () => {
+        DOM.scoresSelector.textContent = scores;
+    }
+
+    const checkPair = () => {
+        const openedCards = document.querySelectorAll(`.${DOM.openedCard}`);
+        let IDsOfOpenedCards = [];
```
Comment (zonzujiro) [inline submissions/linkqwd/task-10_memoryPairGame/main.js]: Only constructors can be called from uppercase

File: submissions/linkqwd/task-10_memoryPairGame/main.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+        openedCards.forEach(el => IDsOfOpenedCards.push(+el.dataset.id));
```
Comment (zonzujiro) [inline submissions/linkqwd/task-10_memoryPairGame/main.js]: You can do this with `Array#map`

**Comment by linkqwd** [comment]
Thanks for review.
All done

**Comment by zonzujiro** [comment]
> Thanks for review.
> All done

@linkqwd pls, resolve outdated discussions

File: submissions/linkqwd/task-10_memoryPairGame/main.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    makePreview(settings.animationDuration, () => toggleClass(DOM.openedCard))
+    .then(() => makePreview(settings.prewievDelay, () => toggleClass(DOM.openedCard)))
+    .then(() => makePreview(settings.animationDuration, () => toggleClass(DOM.noClickCard)));
+    
+    function checkPair () {
+        const openedCards = document.querySelectorAll(`.${DOM.openedCard}`);
+        let iDsOfOpenedCards = [...openedCards].map((el) => +el.dataset.id);
+        let isIDsEqual = iDsOfOpenedCards.reduce((stack, value) => (stack === value) ? stack = value : stack = 0);
```
Comment (zonzujiro) [inline submissions/linkqwd/task-10_memoryPairGame/main.js]: If you need to compare two id's - just compare two id's. No need in `reduce`, `map`, etc. But this is on future.

Comment (linkqwd) [reply submissions/linkqwd/task-10_memoryPairGame/main.js]: The point of this is that there can be more than two id's.
In game options it is possible to specify how many cards would be in one pair `numberOfCardsToGuess: 4`
So I had to use some solution like this.

**Pull request merged**